tools/refresh-keyrings.sh

```diff
@@ -25,24 +54,9 @@ generate_keyring() {
         --recv-keys $KEYS
 }
 
-if [ $(lsb_release -sc) != "trusty" ]; then
-    echo "ERROR: must run on trusty to ensure compatibility"
-    exit 1
-fi
-
-TARGET_DIR="$1"
-
-KEYRING_ESM="ubuntu-esm-v2-keyring.gpg"
-KEYRING_FIPS="ubuntu-fips-keyring.gpg"
-KEYRING_FIPS_UPDATES="ubuntu-fips-updates-keyring.gpg"
-KEYRING_CC="ubuntu-cc-keyring.gpg"
-
-KEYS_ESM="4067E40313CB4B13"
-KEYS_FIPS="C1997C40EDE22758"
-KEYS_FIPS_UPDATES="C1997C40EDE22758"
-KEYS_CC="3A186E733F491C46"
+UA_KEYRING="ubuntu-advantage-keyring.gpg"
 
-generate_keyring "$TARGET_DIR/$KEYRING_ESM" "$KEYS_ESM"
-generate_keyring "$TARGET_DIR/$KEYRING_FIPS" "$KEYS_FIPS"
-generate_keyring "$TARGET_DIR/$KEYRING_FIPS_UPDATES" "$KEYS_FIPS_UPDATES"
-generate_keyring "$TARGET_DIR/$KEYRING_CC" "$KEYS_CC"
+APT_KEYS=`cat $MACHINE_TOKEN | json_pp | awk -F '"' '/aptKey.*/{printf $4 " "}'`
```

Comment:
I'm not convinced we should be making this change; we want an easily reviewable change process for the keys we ship, and having them hard-coded in this file makes that much easier, IMO.

Also: a token doesn't necessarily have access to all the entitlements, and we may want to ship more keys than just the current `aptKey` values, to support key transitions. So I'm not sure it's possible to fully automate this using the contract server, which means we need the manual key handling anyway.

(A potential future improvement would be to check that the set of keys in the keyring after generation is a super-set of the set of keys specified by the `aptKey` directives, but that's very much a nice-to-have so if you think that's worthwhile let's file an issue for it and prioritise it separately.)

Reply:
+1 I think you sold me on the " values, to support key transitions" part. I'll add hardcoded values then for each series here. And we can validate with this script that the keyring generated contains the set of static keys we add here